### ISSUE TITLE
Had to nearly rewrite Infractions

### ISSUE THREAD

**Issue by godleydemon** [NamelessMC/Nameless#502]
Was looking to use Infractions with BungeeUtilisals and it didn't work. After much much trouble shooting I realized it was because I was using PHP7 and the code for the BungeeUtilisals section was programmed with PHP5 in mind. As msql_connect and it's other functions are deprecated in PHP7. So this is what I ended up writing up to make it display the BungeeUtilisals. It's ugly as all get out, but it works. Would love to see a different solution to this that's a little more elegant if possible.

```
<?php 
/* 
 *	Made by Samerton
 *  http://worldscapemc.co.uk
 *
 *  License: MIT
 */

// Infractions addon page
$page = 'Infractions'; // for navbar

// Ensure the addon is enabled
if(!in_array('Infractions', $enabled_addon_pages)){
	// Not enabled, redirect to homepage
	echo '<script data-cfasync="false">window.location.replace(\'/\');</script>';
	die();
}

require('core/includes/paginate.php');
require('addons/Infractions/config.php');
require('addons/Infractions/Infractions.php');
require('core/integration/uuid.php');

$pagination = new Pagination();
$timeago = new Timeago();

$inf_plugin = $queries->getWhere('infractions_settings', array('id', '=', 1));

if(!count($inf_plugin)){
	// Need to configure addon
	echo 'Please set up the addon in the AdminCP -> Addons tab.';
	die();
}

$inf_plugin = $inf_plugin[0]->value;

// Redirect to fix pagination if URL does not end in /
if(substr($_SERVER['REQUEST_URI'], -1) !== '/' && !strpos($_SERVER['REQUEST_URI'], '?')){
	echo '<script data-cfasync="false">window.location.replace(\'/infractions/\');</script>';
	die();
}

if(isset($_GET['p'])){
	if(!is_numeric($_GET['p'])){
		Redirect::to('/infractions');
		die();
	} else {
		if($_GET['p'] == 1){ 
			// Avoid bug in pagination class
			Redirect::to('/infractions/');
			die();
		}
		$p = $_GET['p'];
	}
} else {
	$p = 1;
}
?>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="Infractions page for the <?php echo $sitename; ?> community">
    <meta name="author" content="<?php echo $sitename; ?>">
    <meta name="theme-color" content="#454545" />
    <?php if(isset($custom_meta)){ echo $custom_meta; } ?>

    <?php
	$title = $infractions_language['infractions'];
	
	require('core/includes/template/generate.php');
	?>

	<style>
	html {
		overflow-y: scroll;
	}
	</style>
   </head>
   <body>
    <?php
	// Load navbar
	$smarty->display('styles/templates/' . $template . '/navbar.tpl');
	?>

	<div class="container">	
	  <div class="well">
	    <h2><?php echo $infractions_language['infractions']; ?></h2>
	    <?php
	    	$paginate = PaginateArray($p);
			
			$n = $paginate[0];
			$f = $paginate[1];
			
			if(count($all_infractions) > $f){
				$d = $p * 10;
			} else {
				$d = count($all_infractions) - $n;
				$d = $d + $n;
			}
		?>
<?php
ini_set('display_errors', 1);
ini_set('display_startup_errors', 1);
error_reporting(E_ALL);
$con=mysqli_connect("127.0.0.1","Utilisals","█████████","Utilisals");
// Check connection
if (mysqli_connect_errno()) {
  print "Failed to connect to MySQL: " . mysqli_connect_error();
  }
$result = mysqli_query($con, "SELECT * FROM Bans ");?>
<html>
	  <div class="table-responsive">
      <table class="table table-bordered" >
      		<colgroup>
		      <col span="1" style="width: 15%;">
		      <col span="1" style="width: 15%;">
		      <col span="1" style="width: 15%">
			  <?php if($inf_plugin != 'bu'){ ?>
		      <col span="1" style="width: 30%">
		      <col span="1" style="width: 15%">
			  <?php } else { ?>
		      <col span="1" style="width: 45%">
			  <?php } ?>
		      <col span="1" style="width: 10%">
		    </colgroup>
      <thead>
        <tr>
          <th>Username</th>
          <th>StaffMember</th>
          <th>Number</th>
          <th>Reason</th>
          <th>Time</th>
        </tr>
      </thead>
      <tbody>
<?php
          while( $row = mysqli_fetch_assoc( $result ) ){
            $toEcho = 
            "<tr>
              <td><center>{$row['Banned']}</center></td>
              <td><center>{$row['BannedBy']}</center></td>
              <td><center>{$row['Number']}</center></td>
              <td><center>{$row['Reason']}</center></td>";
             $banTime = $row['BanTime'];
             if ($banTime == -1) {
                $toEcho .= "<td><center>Permanent</center></td>";
             } else {
                $toEcho .= "<td><center>{$row['BanTime']}</center></td>";
             }
                $toEcho .= "
            </tr>\n";
                echo $toEcho;
             }
?>
      </tbody>
    </table>
	
	<?php
	// Footer
	require('core/includes/template/footer.php');
	$smarty->display('styles/templates/' . $template . '/footer.tpl');
	
	// Scripts 
	require('core/includes/template/scripts.php');
	?>

  </body>
</html>
```

--edit--
added in a if then else statement for if it's -1 it'll just say Permanent

**Comment by Fiiily** [comment]
Thanks. I'm glad you came along, partner.

**Comment by godleydemon** [comment]
well to be honest, I'm still working a bit on it here. The thing is, I'm not sure how to pull from the other tables and do the switches right.

**Comment by samerton** [comment]
I'll take a look into using BungeeUtilisals with PHP7, however all plugins within the Infractions addon use PDO instead of the mysql functions (which, as you say, are depreciated).

Once the Infractions addon is rewritten as a module for v2, I can also give everything a nice new redesign and make it more efficient.

**Comment by godleydemon** [comment]
thanks alot Samerton, to be honest, I spent way to much time on this today XD. Was trying to convert it straight out and was having issues since I'm so used to php5 and not php7. So I'm unfamiliar with some of the changes that were made between versions and the usage of mysqli. That, and every fiber of my being was screaming do it in mysqli, even though a small little section of my brain started nudging me, "maybe we should try and do this in PDO" lol.

When running everything out of the gate though, I wasn't getting anything appearing within the Infractions page. So that was the entire point of trying to debug it and try to figure out why it wasn't working. That's when I found out it wasn't working because it was using some older functions for php5 with mysql_connect and such. Or it could be that I've really not had enough coffee all day and am entirely way to stressed out lol
